jupyter_client/asynchronous/client.py

```diff
@@ -0,0 +1,391 @@
+"""Implements an async kernel client"""
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+from __future__ import print_function
+
+from functools import partial
+from getpass import getpass
+try:
+    from queue import Empty  # Python 3
+except ImportError:
+    from Queue import Empty  # Python 2
+import sys
+import time
+
+import zmq
+import zmq.asyncio
+import asyncio
+
+from traitlets import (Type, Instance)
+from jupyter_client.channels import HBChannel
+from jupyter_client.client import KernelClient
+from .channels import ZMQSocketChannel
+
+try:
+    monotonic = time.monotonic
+except AttributeError:
```

Comment:
Can drop for python 3 only (next non-patch release will probably not include python 2 support regardless)